ejercicios lunes, no entiendo practicamente nada

aidagarcd opened the pull request:
aida garcia ejercicios lunes, no he entendido aunque comprendo la estructura

aidagarcd committed:
ejercicios lunes, no entiendo practicamente nada

sh-ih commented:
Para usar una función con map, filter o reduce, primero necesitas escribir el código de tu función y después de eso ya la puedes usar.

Por ejemplo, en el Challenge 1:
For example, let's look at words 1 through 10.
Tienes una lista llamada new_text. Para ver las primeras 10 palabras, puedes usar indexing:
`new_list[:10]`
Que te da como resultado:
`['PROPHET\n\n|Almustafa,', 'the{7}', 'chosen', 'and', 'the\nbeloved,', 'who', 'was', 'a', 'dawn', 'unto']`

The next step is to create a function that will remove references.
We will do this by splitting the string on the { character and keeping only the part before this character. Write your function below.

Recuerda que una función inicia con la palabra def seguida del nombre de la función y los argumentos que recibirá (en caso de que sea una función que recibe argumentos).
Después debes añadir el código que realiza la acción que deseas de tu función.
Finalizas tu función con return y la variable que es el output/resultado de tu función

Por ejemplo:
```
def sum(a,b):
	suma = a+b
	return suma
```

En el caso del Challenge 1, tienes esto como código inicial:
```
def reference(x):
    '''
    Input: A string
    Output: The string with references removed
    
    Example:
    Input: 'the{7}'
    Output: 'the'
    '''
    
    # your code here
    # Aquí es donde debes añadir tu código
    splitted_string = x.split('{')
    string_before_split = splitted_string[0]
    
    return string_before_split
```

Y ya que tienes tu función, la puedes usar:
```
l = list(map(reference,prophet))
print(l)
```

En el código que dejaste, no tenías código para definir lo que sucede adentro de la función, así que no puedes usarla en map
```
def reference(x):
    '''
    Input: A string
    Output: The string with references removed
    
    Example:
    Input: 'the{7}'
    Output: 'the'
    '''
l = map(list(reference, x.split(("{}"))))    
```

Y la sintaxis de map es map(función, iterable): en este caso sería `map(reference,prophet)`. 
Y esto lo convertimos en lista, para poder ver los elementos: `list(map(reference,prophet))`

Revisa esto y aplícalo en el resto de los ejercicios. Avísame si tienes otras preguntas o quieres que lo revisemos con más detalle

aidagarcd commented:
Estoy perdida, necesitaría algo tipo una tutoría

El mar, 17 oct 2023 a las 5:54, Sandra Hernández ***@***.***>)
escribió:

> Para usar una función con map, filter o reduce, primero necesitas escribir
> el código de tu función y después de eso ya la puedes usar.
>
> Por ejemplo, en el Challenge 1:
> For example, let's look at words 1 through 10.
> Tienes una lista llamada new_text. Para ver las primeras 10 palabras,
> puedes usar indexing:
> new_list[:10]
> Que te da como resultado:
> ['PROPHET\n\n|Almustafa,', 'the{7}', 'chosen', 'and', 'the\nbeloved,',
> 'who', 'was', 'a', 'dawn', 'unto']
>
> The next step is to create a function that will remove references.
> We will do this by splitting the string on the { character and keeping
> only the part before this character. Write your function below.
>
> Recuerda que una función inicia con la palabra def seguida del nombre de
> la función y los argumentos que recibirá (en caso de que sea una función
> que recibe argumentos).
> Después debes añadir el código que realiza la acción que deseas de tu
> función.
> Finalizas tu función con return y la variable que es el output/resultado
> de tu función
>
> Por ejemplo:
>
> def sum(a,b):
> 	suma = a+b
> 	return suma
>
> En el caso del Challenge 1, tienes esto como código inicial:
>
> def reference(x):
>     '''
>     Input: A string
>     Output: The string with references removed
>
>     Example:
>     Input: 'the{7}'
>     Output: 'the'
>     '''
>
>     # your code here
>     # Aquí es donde debes añadir tu código
>     splitted_string = x.split('{')
>     string_before_split = splitted_string[0]
>
>     return string_before_split
>
> Y ya que tienes tu función, la puedes usar:
>
> l = list(map(reference,prophet))
> print(l)
>
> En el código que dejaste, no tenías código para definir lo que sucede
> adentro de la función, así que no puedes usarla en map
>
> def reference(x):
>     '''
>     Input: A string
>     Output: The string with references removed
>
>     Example:
>     Input: 'the{7}'
>     Output: 'the'
>     '''
> l = map(list(reference, x.split(("{}"))))
>
> Y la sintaxis de map es map(función, iterable): en este caso sería
> map(reference,prophet).
> Y esto lo convertimos en lista, para poder ver los elementos:
> list(map(reference,prophet))
>
> Revisa esto y aplícalo en el resto de los ejercicios. Avísame si tienes
> otras preguntas o quieres que lo revisemos con más detalle
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ironhack-data-bcn-oct-2023/lab-map-filter-reduce/pull/7#issuecomment-1765621874>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5D6VBC3P7MIKWGRVV5JHRTX7X6O5AVCNFSM6AAAAAA6CP5YM2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRVGYZDCOBXGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <Ironhack-data-bcn-oct-2023/lab-map-filter-reduce/pull/7/c1765621874@
> github.com>
>
